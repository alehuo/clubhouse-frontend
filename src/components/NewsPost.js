import React from "react";
import { Button, Panel } from "react-bootstrap";
import { Link } from "react-router-dom";
import FontAwesome from "react-fontawesome";

const NewsPost = ({
  title,
  message,
  author,
  date,
  onDelete,
  onEdit,
  hasEditDeletePermissions
}) => {
  return (
    <Panel bsStyle="primary">
      <Panel.Heading>
        <Panel.Title>
          <b>{title}</b>
          {hasEditDeletePermissions && (
            <React.Fragment>
              {"   "}
              <Button bsStyle="danger" onClick={onDelete}>
                <FontAwesome name="trash" /> Delete
              </Button>
              {"   "}
              <Button bsStyle="success" onClick={onEdit}>
                <FontAwesome name="edit" /> Edit
              </Button>
            </React.Fragment>
          )}
        </Panel.Title>
      </Panel.Heading>
      <Panel.Body>{message}</Panel.Body>
      <Panel.Footer>
        By <Link to={"/user/" + author.id}>{author.name}</Link> on <i>{date}</i>
      </Panel.Footer>
    </Panel>
  );
};

export default NewsPost;
