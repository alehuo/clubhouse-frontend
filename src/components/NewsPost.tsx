import React from "react";
import { Button, Panel } from "react-bootstrap";
import FontAwesome from "react-fontawesome";

const NewsPost: React.SFC<any> = ({
  title,
  message,
  author,
  date,
  onDelete,
  onEdit,
  hasEditDeletePermissions
}) => (
  <Panel bsStyle="primary">
    <Panel.Heading>
      <Panel.Title>
        <b>{title}</b>
        {hasEditDeletePermissions && (
          <React.Fragment>
            {"   "}
            <Button bsStyle="danger" onClick={onDelete}>
              <FontAwesome name="trash" /> Delete
            </Button>
            {"   "}
            <Button bsStyle="success" onClick={onEdit}>
              <FontAwesome name="edit" /> Edit
            </Button>
          </React.Fragment>
        )}
      </Panel.Title>
    </Panel.Heading>
    <Panel.Body>{message}</Panel.Body>
    <Panel.Footer>
      By <strong>{author.name}</strong> on <i>{date}</i>
    </Panel.Footer>
  </Panel>
);

export default NewsPost;
