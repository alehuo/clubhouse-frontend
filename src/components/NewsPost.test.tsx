import { shallow } from "enzyme";
import React from "react";
import NewsPost from "./NewsPost";

it("NewsPost matches snapshot", () => {
  const cmpnt = shallow(
    <NewsPost
      author={1}
      message="Hello World!"
      title="Test title"
      onDelete={() => null}
      onEdit={() => null}
      hasEditDeletePermissions={false}
      date="2019-12-30 12:00:00"
    />,
  );
  expect(cmpnt).toMatchSnapshot();
});

it("NewsPost matches snapshot (has permissions)", () => {
  const cmpnt = shallow(
    <NewsPost
      author={1}
      message="Hello World!"
      title="Test title"
      onDelete={() => null}
      onEdit={() => null}
      hasEditDeletePermissions={true}
      date="2019-12-30 13:00:00"
    />,
  );
  expect(cmpnt).toMatchSnapshot();
});
