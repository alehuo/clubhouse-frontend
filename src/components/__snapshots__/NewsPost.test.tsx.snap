// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsPost matches snapshot (has permissions) 1`] = `ShallowWrapper {}`;

exports[`NewsPost matches snapshot 1`] = `ShallowWrapper {}`;
